src/plugins/visualizations/public/react_embeddable/visualize_embeddable.tsx

```diff
@@ -0,0 +1,532 @@
+          const adapters = inspectorAdapters$.getValue();
+          if (!adapters) return;
+          const inspector = getInspector();
+          if (!inspector.isAvailable(adapters)) return;
+          return getInspector().open(adapters, {
+            title:
+              titlesApi.panelTitle?.getValue() ||
+              i18n.translate('visualizations.embeddable.inspectorTitle', {
+                defaultMessage: 'Inspector',
+              }),
+          });
+        },
+        // Library transforms
+        saveToLibrary: (newTitle: string) => {
+          titlesApi.setPanelTitle(newTitle);
+          const { rawState, references } = serializeState({
+            serializedVis: vis$.getValue().serialize(),
+            titles: {
+              ...serializeTitles(),
+              title: newTitle,
+            },
+          });
+          return saveToLibrary({
+            uiState: vis$.getValue().uiState,
+            rawState: rawState as VisualizeOutputState,
+            references,
+          });
+        },
+        canLinkToLibrary: () => !state.linkedToLibrary,
+        canUnlinkFromLibrary: () => !!state.linkedToLibrary,
+        checkForDuplicateTitle: () => false, // Handled by saveToLibrary action
+        getByValueState: () => ({
+          savedVis: vis$.getValue().serialize(),
+          ...serializeTitles(),
+        }),
+        getByReferenceState: (libraryId) =>
+          serializeState({
+            serializedVis: vis$.getValue().serialize(),
+            titles: serializeTitles(),
+            id: libraryId,
+            linkedToLibrary: true,
+          }).rawState,
+      },
+      {
+        ...titleComparators,
+        ...customTimeRangeComparators,
+        ...(dynamicActionsApi?.dynamicActionsComparator ?? {
+          enhancements: getUnchangingComparator(),
+        }),
+        serializedVis: [
+          serializedVis$,
+          (value) => {
+            serializedVis$.next(value);
+          },
+          (a, b) => {
+            const visA = a
+              ? {
+                  ...omitBy(a, isEmpty),
+                  data: omitBy(a.data, isNil),
+                  params: omitBy(a.params, isNil),
+                }
+              : {};
+            const visB = b
+              ? {
+                  ...omitBy(b, isEmpty),
+                  data: omitBy(b.data, isNil),
+                  params: omitBy(b.params, isNil),
+                }
+              : {};
+            return isEqual(visA, visB);
+          },
+        ],
+        savedObjectId: [
+          savedObjectId$,
+          (value) => savedObjectId$.next(value),
+          (a, b) => {
+            if (!a && !b) return true;
+            return a === b;
+          },
+        ],
+        savedObjectProperties: getUnchangingComparator(),
+        linkedToLibrary: [linkedToLibrary$, (value) => linkedToLibrary$.next(value)],
+      }
+    );
+
+    const fetchSubscription = fetch$(api)
+      .pipe(
+        switchMap((data) => {
```

Comment:
Instead of returning an IIFE, you can just make the switchMap function async and remove the IIFE.

Reply:
@ThomThomson do you remember the reason you told me not to do this and top use an IIFE instead?

Reply:
Hum - I think I was under the impression that you needed to put a `promise` type directly in the first arg slot of the `switchMap` and that a function that returns a promise was not sufficient. I might have been mistaken about this as I'm now not finding where I got that information.

@nreese - if you're certain this syntax works it is definitely preferable.

Reply:
Changing this doesn't seem to break anything so I'm going to push it, but I'll revert if @ThomThomson can confirm the IIFE implementation was necessary

Reply:
> if you're certain this syntax works it is definitely preferable

The syntax works, here is a simpler example of where switchMap is used with an async callback https://github.com/elastic/kibana/blob/main/src/plugins/controls/public/react_controls/controls/data_controls/initialize_data_control.ts#L84.